### ISSUE TITLE
Secondary Targeting Methods

### ISSUE THREAD

**Issue by MrNiceUK** [X2CommunityCore/X2WOTCCommunityHighlander#751]
As I'm sure some of you already know, this is in the ability template:
```UC
var class<X2TargetingMethod>    TargetingMethod;						// UI interaction class. Specifies how the target is actually selected by the user
var class<X2TargetingMethod>	SecondaryTargetingMethod;			// Same as Targeting Method, but this gets used if we want a separate target method after we select/confirm an ability.
```
There is some commented out code in some hud UI for it, and the tactical input has a state for it, but it is currently non-functional. The *intent* is when an ability activation wants two *distinct* pieces of information from the player, like say, a destination and then a direction. Any interest in resurrecting this?

**Comment by Iridar** [comment]
Related to Pull Request #762

**Comment by Iridar** [comment]
Definitely. @E3245 and I have already tried our hand at doing something like this. We wanted to implement line-targeting, but with the line having an arbitrary starting point, not the ability owner's tile position. We hit a native wall at some point, maybe it was `XCGS_Ability::GatherAbilityTargets()`, maybe not. I wasn't aware that Firaxis had some unused variables for this, but I'm not sure they would help. 

I recall putting high hopes on #918 to let us bypass that native wall. Gotta reinvestigate that.

**Comment by Xymanek** [comment]
@Iridar #918 has been merged, time to investigate this again?

**Comment by Iridar** [comment]
> @Iridar #918 has been merged, time to investigate this again?

Uhh I'm in the middle of something right meow, hopefully I'll be a bit more free next week to revisit this.
